src/core/thread/mle.cpp

```diff
@@ -692,10 +692,17 @@ void Mle::SetStateChild(uint16_t aRloc16)
 #endif
 
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-    if (Get<Mac::Mac>().IsCslEnabled())
+    if (Get<Mac::Mac>().IsCslCapable())
```

Comment:
I think the name `IsCslCapable` is a little confusing since `RxOffWhenIdle` is apparently a necessity for CSL capability. What about we change the code like this, still using the `IsCslEnabled`:
```
if (Get<Mac::Mac>().IsCslEnabled())
{
    IgnoreError(Get<Radio>().EnableCsl(Get<Mac::Mac>().GetCslPeriod(), GetParent().GetRloc16(),
                                           &GetParent().GetExtAddress()));
}
else if (!IsRxOnWhenIdle())
{
    IgnoreError(Get<Radio>().EnableCsl(0, GetParent().GetRloc16(), &GetParent().GetExtAddress()));
}
```
I think the code is the same in terms of logic. Thoughts? @edmont @canisLupus1313

Reply:
@Irving-cl Unfortunatelly the IsCslEnabled method is not returning real state of CSL. The flag "RxOnWhenIdle" is stored in two places (MLE)[https://github.com/openthread/openthread/blob/main/src/core/thread/mle.hpp#L1611] and (MAC)[https://github.com/openthread/openthread/blob/main/src/core/mac/mac.hpp#L855]. During transitions between SSED -> MED and MED -> SSED those two aren't synchronized. 
The synchronization happens in sending child update request (here)[https://github.com/openthread/openthread/blob/main/src/core/thread/mle.cpp#L2366] or on reception of child update request reply (here)[https://github.com/openthread/openthread/blob/main/src/core/thread/mle.cpp#L3840]. Ofcourse both of those places are already to late to EnableCsl.
That's why I added extra method which is independent from `IsRxOnWhenIdle` from MAC.